packages/syft/src/syft/node/node.py

```diff
@@ -1243,21 +1243,76 @@ def add_queueitem_to_queue(
             return result
         return job
 
+    def _can_enqueue_user_code_job(
+        self, user_code_id: UID, credentials: SyftVerifyKey
+    ) -> bool:
+        role = self.get_role_for_credentials(credentials=credentials)
+        context = AuthedServiceContext(node=self, credentials=credentials, role=role)
+
+        user_code_service = self.get_service("usercodeservice")
+        user_code = user_code_service.get_by_uid(context=context, uid=user_code_id)
+
+        # We cannot enqueue job if there are no permissions to execute job action
+        override_execution_permission = (
+            context.has_execute_permissions or context.role == ServiceRole.ADMIN
+        )
+        return override_execution_permission or user_code_service.is_execution_allowed(
+            user_code, context, user_code.output_policy
+        )
+
+    def _get_existing_user_code_jobs(
+        self, user_code_id: UID, credentials: SyftVerifyKey
+    ) -> Union[List[Job], SyftError]:
+        role = self.get_role_for_credentials(credentials=credentials)
+        context = AuthedServiceContext(node=self, credentials=credentials, role=role)
+
+        job_service = self.get_service("jobservice")
+        return job_service.get_by_user_code_id(
+            context=context, user_code_id=user_code_id
+        )
+
     def add_api_call_to_queue(self, api_call, parent_job_id=None):
         unsigned_call = api_call
         if isinstance(api_call, SignedSyftAPICall):
             unsigned_call = api_call.message
 
         is_user_code = unsigned_call.path == "code.call"
 
-        service, method = unsigned_call.path.split(".")
+        service_str, method_str = unsigned_call.path.split(".")
 
         action = None
         if is_user_code:
             action = Action.from_api_call(unsigned_call)
+            if self.node_side_type == NodeSideType.LOW_SIDE:
```

Comment:
what is the reason to not handle this within the service? This seems like its service specific logic to me

Reply:
Leaving this here for now, see https://github.com/OpenMined/PySyft/pull/8377#discussion_r1458603133